### PR TITLE
planner: performance optimization for plan-cache

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #36598

Problem Summary: planner: performance optimization for plan-cache

### What is changed and how it works?

planner: performance optimization for plan-cache

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/plan_cache_utils.go [hunk 1]
```diff
@@ -521,52 +488,59 @@ func tableStatsVersionForPlanCache(tStats *statistics.Table) (tableStatsVer uint
...
-		sctx.GetSessionVars().StmtCtx.SetSkipPlanCache(errors.New(checker.unCacheableReason))
+		for _, node := range stmt.QueryFeatures.limits {
+			if node.Count != nil {
+				if count, isParamMarker := node.Count.(*driver.ParamMarkerExpr); isParamMarker {
+					typeExpected, val := CheckParamTypeInt64orUint64(count)
+					if !typeExpected {
+						sctx.GetSessionVars().StmtCtx.SetSkipPlanCache(errors.New("unexpected value after LIMIT"))
+						break
+					}
+					if val > 10000 {
```
Comment (hawkingrei) [inline planner/core/plan_cache_utils.go:515]: We make it as a const variable.

Comment (qw4990) [reply planner/core/plan_cache_utils.go]: Updated. PTAL.

**Review by xuyifangreeneyes — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**